lib/enum/basic_enum.rb

```diff
@@ -71,13 +81,18 @@ def drop_until_its_hot
   # http://ruby-doc.org/core-2.1.4/Enumerable.html#method-i-each_slice
   # Tip: After .each_slice chain on .to_a to turn the returned Enumerator into an array
   def groups_of(num)
+    collection.each_slice(num).to_a
   end
 
 
   # Refer to each_with_object
   # http://ruby-doc.org/core-2.1.4/Enumerable.html#method-i-each_with_object
   # Tip: You'll have to do .each_with_object(Hash.new(0)) to create a hash to start with. Do not save the hash in a variable. You may use the incrementor method here.
+
+  # Perhaps it's because they go in to worse / not the ruby way to do it, but I found this a better resource for showing how this formula works http://batsov.com/articles/2013/12/04/using-rubys-each-with-object/
+
   def element_frequency_count
+    collection.each_with_object(Hash.new(0)) {|element, occurrence| occurrence[element] += 1}
```

Comment:
each_with_object takes two parameters, but I wouldn't call them `|element, occurrence|`. It's more accurate to say `|key, new_hash|` where new_hash is the hash you created with Hash.new(0). It doesn't make sense to say `occurrence[element] += 1` but it does make sense to create new keys in our hash, like this `new_hash[key] += 1`